src/verifast.ml

```diff
@@ -650,13 +670,20 @@ module VerifyProgram(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
               consume_chunk rules h [] [] [] l (uninit_array_pred_symb, true) [] real_unit real_unit_pat (Some 2) [TermPat addr; TermPat n; dummypat] @@ fun _ h _ _ _ _ _ _ ->
               cont h env
           | None -> 
-            let verify_dtor_call = verify_dtor_call (pn, ilist) leminfo funcmap predinstmap sizemap tenv ghostenv h env addr None in
-            consume_cxx_object l real_unit_pat addr t verify_dtor_call false h env @@ fun h env ->
-              begin match t with 
+            begin match t with 
               | StructType name ->
-                  let new_block_symb = get_pred_symb_from_map name new_block_pred_map in
-                  consume_chunk rules h [] [] [] l (new_block_symb, true) [] real_unit real_unit_pat (Some 1) [TermPat addr] @@ fun _ h _ _ _ _ _ _ -> 
-                  cont h env
+                let verify_dtor_call = verify_dtor_call (pn, ilist) leminfo funcmap predinstmap sizemap tenv ghostenv h env addr None in
+                let new_block_symb = get_pred_symb_from_map name new_block_pred_map in
+                begin fun cont ->
+                  if is_polymorphic_struct name then
+                    consume_chunk rules h [] [] [] l (new_block_symb, true) [] real_unit real_unit_pat (Some 1) [TermPat addr; dummypat] @@ fun _ h _ [_; type_info] _ _ _ _ -> 
+                    consume_chunk rules h [] [] [] l (get_pred_symb_from_map name cxx_vtype_map, true) [] real_unit real_unit_pat (Some 1) [TermPat addr; TermPat type_info] @@ fun _ _ _ _ _ _ _ _ ->
+                    cont h
+                  else
+                    consume_chunk rules h [] [] [] l (new_block_symb, true) [] real_unit real_unit_pat (Some 1) [TermPat addr] @@ fun _ h _ _ _ _ _ _ -> 
+                    cont h
+                end @@ fun h ->
+                consume_cxx_object l real_unit_pat addr t verify_dtor_call false h env cont
               | _ -> 
                 consume_chunk rules h [] [] [] l (get_pred_symb "new_block", true) [] real_unit real_unit_pat (Some 1) [TermPat addr; TermPat (sizeof l t)] @@ fun _ h _ _ _ _ _ _ -> 

```

Comment:
If the type does not have a pointee chunk and is not a struct, you consume only the new_block?

Reply:
An example of such a type is `int32_t` in targetless mode, which is represented by an `integer_` chunk.